hbase-server/src/main/java/org/apache/hadoop/hbase/coprocessor/RegionObserver.java

```diff
@@ -216,7 +214,7 @@ void preCompactSelection(final ObserverContext<RegionCoprocessorEnvironment> c,
    * @param request custom compaction request
    */
   void postCompactSelection(final ObserverContext<RegionCoprocessorEnvironment> c,
-      final Store store, final ImmutableList<StoreFile> selected, CompactionRequest request);
+      final Store store, final List<StoreFile> selected, CompactionRequest request);
```

Comment:
RegionObserver API fix. This method should not receive Guava's ImmutableList as type, it should simply be List. Using ImmutableList to represent the fact that this list is, and should be, immutable, is nice, but binds a LimitedPrivate interface to the vagaries of Guava's frequent breaking changes as it itself evolves.